### PR TITLE
Document progress indicators

### LINKED ISSUE #3055: Document how to create an indeterminate progress indicator
You're supposed to use a null value, which is written in a comment but not included in the dartdoc.

### PR DESCRIPTION
**Pull request by abarth**
Specifically, explain how to create both determinate and indeterminate progress
indicators.

Fixes #3055

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @qchong

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 1]
```diff
@@ -14,13 +14,34 @@ const double _kCircularProgressIndicatorStrokeWidth = 4.0;
 
 // TODO(hansmuller): implement the support for buffer indicator
 
+/// An abstract material design progress indicator.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/progress_indicator.dart]: the indicator itself isn't abstract. A base class for material design progress indicators, maybe?

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 2]
```diff
@@ -14,13 +14,34 @@ const double _kCircularProgressIndicatorStrokeWidth = 4.0;
...
+  /// The [value] argument can be either null (cooresponding to an indeterminate
+  /// progress indcator) or non-null (cooresponding to a determinate progress
+  /// indicator). See [value] for details.
   ProgressIndicator({
     Key key,
     this.value
   }) : super(key: key);
 
-  final double value; // Null for non-determinate progress indicator.
+  /// If non-null, the value of this progress indicator with 0.0 cooresponding
```
Comment (Hixie) [inline packages/flutter/lib/src/material/progress_indicator.dart]: corresponding

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 3]
```diff
@@ -14,13 +14,34 @@ const double _kCircularProgressIndicatorStrokeWidth = 4.0;
...
+  /// to no progress having been made and 1.0 cooresponding to all the progress
```
Comment (Hixie) [inline packages/flutter/lib/src/material/progress_indicator.dart]: ditto

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 4]
```diff
@@ -14,13 +14,34 @@ const double _kCircularProgressIndicatorStrokeWidth = 4.0;
...
+  /// having been made.
+  ///
+  /// If null, this progress indicator is non-determinate, which means the
```
Comment (Hixie) [inline packages/flutter/lib/src/material/progress_indicator.dart]: indeterminate?

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 5]
```diff
@@ -14,13 +14,34 @@ const double _kCircularProgressIndicatorStrokeWidth = 4.0;
...
+  /// indicator displays a predetermined animation that does not indicator how
+  /// much actually progress is being made.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/progress_indicator.dart]: actual

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 6]
```diff
@@ -74,7 +95,30 @@ class _LinearProgressIndicatorPainter extends CustomPainter {
...
+/// See also:
+///
+///  * [CircularProgressIndicator]
+///  * <https://www.google.com/design/spec/components/progress-activity.html#progress-activity-types-of-indicators>
 class LinearProgressIndicator extends ProgressIndicator {
+  /// Creates a linear progress indicator.
+  ///
+  /// The [value] argument can be either null (cooresponding to an indeterminate
+  /// progress indcator) or non-null (cooresponding to a determinate progress
+  /// indicator). See [value] for details.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/progress_indicator.dart]: (same fixes as previous identical paragraph)

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 7]
```diff
@@ -74,7 +95,30 @@ class _LinearProgressIndicatorPainter extends CustomPainter {
   }
 }
 
+/// An material design linear progress indicator.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/progress_indicator.dart]: An->a

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 8]
```diff
@@ -196,7 +240,30 @@ class _CircularProgressIndicatorPainter extends CustomPainter {
   }
 }
 
+/// An material design circular progress indicator.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/progress_indicator.dart]: An->a

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 9]
```diff
@@ -196,7 +240,30 @@ class _CircularProgressIndicatorPainter extends CustomPainter {
...
+/// See also:
+///
+///  * [LinearProgressIndicator]
+///  * <https://www.google.com/design/spec/components/progress-activity.html#progress-activity-types-of-indicators>
 class CircularProgressIndicator extends ProgressIndicator {
+  /// Creates a circular progress indicator.
+  ///
+  /// The [value] argument can be either null (cooresponding to an indeterminate
+  /// progress indcator) or non-null (cooresponding to a determinate progress
+  /// indicator). See [value] for details.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/progress_indicator.dart]: ditto

**Comment by Hixie** [comment]
LGTM

**Pull request merged**